src/Sulu/Bundle/ContentBundle/Resources/public/js/components/ckeditor-internal-link/main.js

```diff
@@ -0,0 +1,215 @@
+                                    this.sandbox.emit('husky.overlay.internal-link.slide-to', 0);
+
+                                    return false;
+                                }.bind(this)
+                            }
+                        ]
+                    }
+                }
+            ]).then(function() {
+                this.sandbox.form.create(formSelector).initialized.then(function() {
+                    this.setData(this.options.link).then(this.initializeFormComponents.bind(this));
+
+                    this.bindDomEvents();
+                }.bind(this));
+            }.bind(this));
+        },
+
+        initializeFormComponents: function() {
+            this.sandbox.start(
+                [
+                    {
+                        name: 'loader@husky',
+                        options: {
+                            el: this.$find('.loader')
+                        }
+                    },
+                    {
+                        name: 'content-datasource@sulucontent',
+                        options: {
+                            el: '#href-select',
+                            selected: this.options.link.href,
+                            webspace: this.options.webspace,
+                            locale: this.options.locale,
+                            selectedUrl: '/admin/api/nodes/{datasource}?tree=true&webspace={webspace}&language={locale}&fields=title,order&webspace-nodes=all',
+                            rootUrl: '/admin/api/nodes?webspace={webspace}&language={locale}&fields=title,order&webspace-nodes=all',
+                            resultKey: 'nodes',
+                            instanceName: 'internal-link',
+                            selectCallback: function(id, path, title) {
+                                var $value = $('#internal-link-href-value');
+                                $value.val(title);
+                                $('#internal-link-href-button-clear').show();
+
+                                this.href = id;
+                                this.hrefTitle = title;
+                                this.sandbox.emit('husky.overlay.internal-link.slide-to', 0);
+                            }.bind(this)
+                        }
+                    }
+                ]
+            ).then(function() {
+                if (!this.options.link.href) {
+                    this.showHrefInput();
+
+                    return;
+                }
+
+                this.sandbox.once('husky.column-navigation.smart-content-internal-link.loaded', function() {
+                    this.sandbox.emit('husky.column-navigation.smart-content-internal-link.get-breadcrumb', function(breadcrumb) {
+                        if (breadcrumb.length === 0) {
+                            this.showHrefInput();
+
+                            return;
+                        }
+
+                        this.setHref(breadcrumb[breadcrumb.length - 1]);
```

Comment:
Do we have to load that via the breadcrumbs? Isn't there a more elegant version?